### PR TITLE
adding transform method for sf objects #1284 

### PR DESCRIPTION
**Pull request by lcgodoy**

### CODE REVIEW

**Comment by etiennebr** [comment]
Linking to #1284 (don't know why github didn't pick it up from the title)

**Comment by etiennebr** [comment]
Would calling `NextMethod()` work? Using `NextMethod` would probably make it easier to maintain.

I also think it would be good to add a few tests to complete code coverage. Also test edge cases, such as `transform(sf, geometry = 1)` and maybe make the behavior consistent with `mutate.sf` (which creates an invalid `sf` object in that case :confused:)?

**Comment by lcgodoy** [comment]
@etiennebr what do you expect in this case `transform(sf, geomtry = 1)`? Dropping the geometry column? Not sure if NextMethod would work, but I can try.

File: R/sf.R [hunk 1]
```diff
@@ -453,3 +453,33 @@ st_drop_geometry = function(x) {
...
 }
+
+#' transform method for sf objects
+#' 
+#' Can be used to create or modify variables.
+#' 
+#' @param _data object of class \code{sf}
+#' @param ... Further arguments of the form tag=value
+#'
+#' @export
```
[the remark this answers is not in the document]
Comment (edzer) [reply R/sf.R]: `mutate.sf` uses a trick to export it _only_ when `dplyr` is loaded, in order to make `dplyr` not a dependency of `sf`; since `transform` is a method in `base`, this is not needed here.

[the remark this answers is not in the document]
Comment (lcgodoy) [reply R/sf.R]: thank you for the clarification. I've already modified it in the code. :smile:
